# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/sessionstore/SessionCookies.sys.mjs
# Commit: 5aa3b2f1b5ec
# Full Hash: 5aa3b2f1b5ec8a63fa04f80cf4aed4aac2c6ce22
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 6
# Description:
#   Bug 1839918 - Refactor "cookie-changed" notifications to use nsICookieNotification. r=extension-reviewers,cookie-reviewers,sessionstore-reviewers,bvandersloot,edgul,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185300
# ==============================================================================

diff -r fecbf97cc8db -r 5aa3b2f1b5ec browser/components/sessionstore/SessionCookies.sys.mjs
--- a/browser/components/sessionstore/SessionCookies.sys.mjs	Thu Sep 07 08:22:27 2023 +0000
+++ b/browser/components/sessionstore/SessionCookies.sys.mjs	Thu Sep 07 08:25:54 2023 +0000
@@ -91,22 +91,32 @@
    * Handles observers notifications that are sent whenever cookies are added,
    * changed, or removed. Ensures that the storage is updated accordingly.
    */
-  observe(subject, topic, data) {
-    switch (data) {
-      case "added":
-        this._addCookie(subject);
+  observe(subject) {
+    let notification = subject.QueryInterface(Ci.nsICookieNotification);
+
+    let {
+      COOKIE_DELETED,
+      COOKIE_ADDED,
+      COOKIE_CHANGED,
+      ALL_COOKIES_CLEARED,
+      COOKIES_BATCH_DELETED,
+    } = Ci.nsICookieNotification;
+
+    switch (notification.action) {
+      case COOKIE_ADDED:
+        this._addCookie(notification.cookie);
         break;
-      case "changed":
-        this._updateCookie(subject);
+      case COOKIE_CHANGED:
+        this._updateCookie(notification.cookie);
         break;
-      case "deleted":
-        this._removeCookie(subject);
+      case COOKIE_DELETED:
+        this._removeCookie(notification.cookie);
         break;
-      case "cleared":
+      case ALL_COOKIES_CLEARED:
         CookieStore.clear();
         break;
-      case "batch-deleted":
-        this._removeCookies(subject);
+      case COOKIES_BATCH_DELETED:
+        this._removeCookies(notification.batchDeletedCookies);
         break;
       default:
         throw new Error("Unhandled session-cookie-changed notification.");